### ISSUE TITLE
Missing files errors when building vim from source...

### ISSUE THREAD

**Issue by zzz0zzz** [vim/vim#3254]
Hi,
I tried to build vim from source using the following script: (taken from https://github.com/Valloric/YouCompleteMe/wiki/Building-Vim-from-source)
```
    cd ~
    sudo apt-get install --yes \
        libncurses5-dev \
        libgnome2-dev \
        libgnomeui-dev \
        libgtk2.0-dev \
        libatk1.0-dev \
        libbonoboui2-dev \
        libcairo2-dev \
        libx11-dev \
        libxpm-dev \
        libxt-dev python3-dev \
        git
    git clone https://github.com/vim/vim.git
    cd vim
    ./configure \
        --with-features=huge \
        --enable-multibyte \
        --enable-python3interp=yes \
        --with-python3-config-dir=/usr/lib/python3.5/config-3.5m-x86_64-linux-gnu \
        --enable-gui=gtk2 \
        --enable-cscope \
        --prefix=/usr/local
  make VIMRUNTIMEDIR=/usr/local/share/vim/vim3.5
  cd ~/vim
  sudo make install
  sudo update-alternatives --install /usr/bin/editor editor /usr/local/bin/vim 1
  sudo update-alternatives --set editor /usr/local/bin/vim
  sudo update-alternatives --install /usr/bin/vi vi /usr/local/bin/vim 1
  sudo update-alternatives --set vi /usr/local/bin/vim
  cd ..
  sudo rm -r vim
```

The command
```
make VIMRUNTIMEDIR=/usr/local/share/vim/vim3.5
```
issues the following complaints:
```
cp: cannot stat '../runtime/doc/xxd-da.1': No such file or directory
cp: cannot stat '../runtime/doc/xxd-da.UTF-8.1': No such file or directory
cp: cannot stat '../runtime/doc/xxd-de.1': No such file or directory
cp: cannot stat '../runtime/doc/xxd-de.UTF-8.1': No such file or directory
sed: can't read ../runtime/doc/evim-da.1: No such file or directory
sed: can't read ../runtime/doc/evim-da.UTF-8.1: No such file or directory
sed: can't read ../runtime/doc/vimtutor-de.1: No such file or directory
sed: can't read ../runtime/doc/evim-de.1: No such file or directory
cp: cannot stat '../runtime/doc/vimdiff-de.1': No such file or directory
sed: can't read ../runtime/doc/vimtutor-de.UTF-8.1: No such file or directory
cp: cannot stat '../runtime/doc/vimdiff-de.UTF-8.1': No such file or directory
sed: can't read ../runtime/doc/evim-de.UTF-8.1: No such file or directory
```

**Comment by zzz0zzz** [comment]
Forgot to say thanks...

**Comment by k-takata** [comment]
> make VIMRUNTIMEDIR=/usr/local/share/vim/vim3.5

Normally you don't need to specify `VIMRUNTIMEDIR`. The default value is defined based on the Vim version (`/usr/local/share/vim/vim81`).

> cp: cannot stat '../runtime/doc/xxd-da.1': No such file or directory
> (snip)

This is because these translations are not available. (German and Danish translations for xxd and etc.)
You can just ignore the messages.

**Comment by chrisbra** [comment]
It's a bit hard to follow the Makefile logic, but perhaps we need a patch like this for installman.sh:
```patch
diff --git a/src/installman.sh b/src/installman.sh
index a2f09ad35..59a161147 100755
--- a/src/installman.sh
+++ b/src/installman.sh
@@ -49,6 +49,7 @@ fi
 if test $what = "install"; then
    # vim.1
    echo installing $destdir/$exename.1
+   test -f $helpsource/vim$langadd.1 &&
    LC_ALL=C sed -e s+/usr/local/lib/vim+$vimloc+ \
           -e s+$vimloc/doc+$helpsubloc+ \
           -e s+$vimloc/print+$printsubloc+ \
@@ -62,26 +63,29 @@ if test $what = "install"; then
           -e s+$vimloc/scripts.vim+$scriptloc/scripts.vim+ \
           -e s+$vimloc/optwin.vim+$scriptloc/optwin.vim+ \
           -e 's+$vimloc/\*.ps+$scriptloc/\*.ps+' \
-          $helpsource/vim$langadd.1 > $destdir/$exename.1
-   chmod $manmod $destdir/$exename.1
+          $helpsource/vim$langadd.1 > $destdir/$exename.1 &&
+      chmod $manmod $destdir/$exename.1

    # vimtutor.1
    echo installing $destdir/$exename""tutor.1
+   test -f $helpsource/vimtutor$langadd.1 &&
    LC_ALL=C sed -e s+/usr/local/lib/vim+$vimloc+ \
           -e s+$vimloc/tutor+$tutorsubloc+ \
-          $helpsource/vimtutor$langadd.1 > $destdir/$exename""tutor.1
+          $helpsource/vimtutor$langadd.1 > $destdir/$exename""tutor.1 &&
    chmod $manmod $destdir/$exename""tutor.1

    # vimdiff.1
    echo installing $destdir/$vimdiffname.1
-   cp $helpsource/vimdiff$langadd.1 $destdir/$vimdiffname.1
+   test -f $helpsource/vimdiff$langadd.1 &&
+   cp $helpsource/vimdiff$langadd.1 $destdir/$vimdiffname.1 &&
    chmod $manmod $destdir/$vimdiffname.1

    # evim.1
    echo installing $destdir/$evimname.1
+   test -f $helpsource/evim$langadd.1 &&
    LC_ALL=C sed -e s+/usr/local/lib/vim+$vimloc+ \
           -e s+$vimloc/evim.vim+$scriptloc/evim.vim+ \
-          $helpsource/evim$langadd.1 > $destdir/$evimname.1
+          $helpsource/evim$langadd.1 > $destdir/$evimname.1 &&
    chmod $manmod $destdir/$evimname.1
 fi

@@ -107,8 +111,9 @@ fi

 if test $what = "xxd"; then
    echo installing $destdir/xxd.1
-   cp $helpsource/xxd$langadd.1 $destdir/xxd.1
-   chmod $manmod $destdir/xxd.1
+   test -f $helpsource/xxd$langadd.1 &&
+      cp $helpsource/xxd$langadd.1 $destdir/xxd.1 &&
+      chmod $manmod $destdir/xxd.1
 fi

 exit $errstatus
```
That checks if the file exists before trying to access it.

**Comment by zzz0zzz** [comment]
It works now, thanks...

**Comment by brammool** [comment]
Christian wrote:

> It's a bit hard to follow the Makefile logic, but perhaps we need a
> patch like this for installman.sh:

Thanks.  I think it's a bit more readable when using an "if" statement.
And then the "installing" message is also suppressed.


-- 
GUARD #1:  What -- a swallow carrying a coconut?
ARTHUR:    It could grip it by the husk!
GUARD #1:  It's not a question of where he grips it!  It's a simple question
           of weight ratios!  A five ounce bird could not carry a 1 pound
           coconut.
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
